patoolib/programs/arc.py

```diff
@@ -15,30 +15,41 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 """Archive commands for the arc program."""
 import os
+from ..util import PatoolError
 
-def extract_arc (archive, compression, cmd, verbosity, interactive, outdir):
+def _add_password_to_options(options, password):
+    """Check password and add it to ARC options."""
+    if password is None:
+        return options
+    if ' ' in password:
+        raise PatoolError("Password for ARC can't contain spaces.")
```

Comment:
I with there was `PatoolValueError`, molding all exceptions as `PatoolError` (there is no other) makes outside handling trickier.  But I guess it should not be this PR to introduce it

Reply:
It's not a big deal to create new exception, much harder is to make correct decision about where to use it in logical and straightforward way. 
I can try to implement it, or later create new PR, from my point of view it's much simpler than adding password 😄

Reply:
I will add it to new PR after merge this.